plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -7239,7 +7239,9 @@ private static File fetchSystemVmKeyFile() {
                                     instanceDisk.setController(DiskControllerType.getType(device.getClass().getSimpleName()).toString());
                                     instanceDisk.setControllerUnit(((VirtualSCSIController) device).getBusNumber());
                                 } else {
-                                    instanceDisk.setController(DiskControllerType.none.toString());
+                                    String msg = String.format("Controller of disk %s is unsupported", instanceDisk.getLabel());
+                                    s_logger.error(msg);
+                                    throw new Exception(msg);
```

Comment:
@DK101010 I think on failure you can decide to fail the entire VM ingestion by (a) returning an empty list and let the user of the `getUnmanageInstanceDisks` method handle the case (when disk list is empty throw an exception), or (b) in the generic exception handler, instead of Exception throw an `UnsupportedOperationException` add another handler for the specific exception. cc @shwstppr

Reply:
@rhtyd @DK101010 same method is used when listUnmanagedINstances API is called so I'm not if we would want the complete API to raise an exception when disks for one particular VM is not found. In my opinion, we can log the error here and return the list of disks for the VM as empty. With an empty list of disks VM import with automatically fail with existing check
https://github.com/apache/cloudstack/blob/master/server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java#L942-L944

If we want to go with the above approach we can clear the `instanceDisks` list variable in the catch block

Reply:
@shwstppr @rhtyd ok, then I will modify the exception msg and clear the list in the catch block. Thanks you two for help.

Reply:
@shwstppr ahh  sorry, I had not checked that you have already changed the msg :D.